lib/csv/table.rb

```diff
@@ -285,7 +291,8 @@ def delete_if(&block)
         @table.delete_if(&block)
       else                                      # by header
         deleted = []
-        headers.each do |header|
+        initial_headers = headers.dup
```

Comment:
Is `dup` needed?

Reply:
I've fixed it(because `dup` had been duplicated in `headers` method).